plugins/woocommerce/tests/legacy/unit-tests/order/class-wc-tests-crud-orders.php

```diff
@@ -611,6 +611,190 @@ public function test_calculate_shipping() {
 		$this->assertEquals( 20, $object->get_shipping_total() );
 	}
 
+	public function test_calculate_shipping_with_shipping_classes() {
+		// Create shipping classes.
+		$shipping_class_1 = wp_insert_term( 'New York', 'product_shipping_class' );
+		$shipping_class_2 = wp_insert_term( 'Florida', 'product_shipping_class' );
+		$shipping_class_3 = wp_insert_term( 'California', 'product_shipping_class' );
+		$shipping_class_4 = wp_insert_term( 'Illinois', 'product_shipping_class' );
+
+		// Create products and assign to each one of them a different shipping class.
+		$product_1 = WC_Helper_Product::create_simple_product( false, array( 'name' => 'Test Shipping Product 1', 'sku' => 'TSSKU1' ) );
+		$product_1->set_shipping_class_id( $shipping_class_1['term_id'] );
+		$product_1->save();
+
+		$product_2 = WC_Helper_Product::create_simple_product( false, array( 'name' => 'Test Shipping Product 2', 'sku' => 'TSSKU2' ) );
+		$product_2->set_shipping_class_id( $shipping_class_2['term_id'] );
+		$product_2->save();
+
+		$product_3 = WC_Helper_Product::create_simple_product( false, array( 'name' => 'Test Shipping Product 3', 'sku' => 'TSSKU3' ) );
+		$product_3->set_shipping_class_id( $shipping_class_3['term_id'] );
+		$product_3->save();
+
+		$product_4 = WC_Helper_Product::create_simple_product( false, array( 'name' => 'Test Shipping Product 4', 'sku' => 'TSSKU4' ) );
+		$product_4->set_shipping_class_id( $shipping_class_4['term_id'] );
+		$product_4->save();
+
+		// Create 3 shipping zones. 2 will have one flat rate each while the last will have 2 flat rates.
+		$zone_1 = new WC_Shipping_Zone();
+		$zone_1->set_zone_name( 'New York' );
+		$zone_1->set_zone_order( 1 );
+		$zone_1->add_location( 'US:NY', 'state' );
+		$instance_id_1 = $zone_1->add_shipping_method( 'flat_rate' );
+		$zone_1->save();
+
+		$zone_2 = new WC_Shipping_Zone();
+		$zone_2->set_zone_name( 'Florida' );
+		$zone_2->set_zone_order( 2 );
+		$zone_2->add_location( 'US:FL', 'state' );
+		$instance_id_2 = $zone_2->add_shipping_method( 'flat_rate' );
+		$zone_2->save();
+
+		$zone_3 = new WC_Shipping_Zone();
+		$zone_3->set_zone_name( 'California and Illinois' );
+		$zone_3->set_zone_order( 3 );
+		$zone_3->add_location( 'US:CA', 'state' );
+		$zone_3->add_location( 'US:IL', 'state' );
+		$instance_id_3 = $zone_3->add_shipping_method( 'flat_rate' );
+		$instance_id_4 = $zone_3->add_shipping_method( 'flat_rate' );
+		$zone_3->save();
+
+		$map = array(
+			$shipping_class_1['term_id'] => $instance_id_1, // NY
+			$shipping_class_2['term_id'] => $instance_id_2, // FL
+			$shipping_class_3['term_id'] => $instance_id_3, // CA
+			$shipping_class_4['term_id'] => $instance_id_4, // IL
+		);
```

Comment:
(1/3) I'm a little concerned about overall readability once we reach the first assertion. What if we instead define the map in this way:

```php
$method_1_cost = '125';
/* Other method costs... */

$map = array(
	$shipping_class_1['term_id'] => array( $instance_id_1, $method_1_cost ),
	/* ... */
);

Reply:
Yea makes sense, done!